### PR TITLE
fix: use provider.request instead of custom ec2 instance

### LINKED ISSUE #1: bug: Uses default credentials instead of provider
Should probably obey `provider: profile` in `serverless.yml` file.

### LINKED ISSUE #4: VPC ENI Cleanup: Error: Missing credentials in config
I am getting this error when running the plugin. Any idea what is wrong? Thanks.

VPC ENI Cleanup: Error: Missing credentials in config
{ Error: connect EHOSTUNREACH 169.254.169.254:80
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
  message: 'Missing credentials in config',
  errno: 'EHOSTUNREACH',
  code: 'CredentialsError',
  syscall: 'connect',
  address: '169.254.169.254',
  port: 80,
  time: 2019-03-06T01:26:18.575Z,
  originalError: 
   { message: 'Could not load credentials from any providers',
     errno: 'EHOSTUNREACH',
     code: 'CredentialsError',
     syscall: 'connect',
     address: '169.254.169.254',
     port: 80,
     time: 2019-03-06T01:26:18.575Z,
     originalError: 
      { message: 'Missing credentials in config',
        errno: 'EHOSTUNREACH',
        code: 'CredentialsError',
        syscall: 'connect',
        address: '169.254.169.254',
        port: 80,
        time: 2019-03-06T01:26:18.574Z,
        originalError: [Object] } } }

### PR DESCRIPTION
**Pull request by onhate**
closes: #1 
closes: #4

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,35 +1,46 @@
...
 
-const d           = require("d")
-    , lazy        = require("d/lazy")
-    , BbPromise   = require("bluebird")
-    , Ec2         = require("aws-sdk/clients/ec2")
-    , { inspect } = require("util");
+const d = require("d")
+	, lazy = require("d/lazy")
+	, BbPromise = require("bluebird")
+	, { inspect } = require("util");
```
Comment (medikoo) [inline index.js:6]: Let's not introduce whitespace changes like that

File: index.js [hunk 2]
```diff
@@ -1,35 +1,46 @@
...
+		return this.provider.request("EC2", "describeNetworkInterfaces", args);
+	}
+
+	detachNetworkInterface(args) {
+		return this.provider.request("EC2", "detachNetworkInterface", args);
+	}
+
+	deleteNetworkInterface(args) {
+		return this.provider.request("EC2", "deleteNetworkInterface", args);
+	}
```
Comment (medikoo) [inline index.js]: Let's not introduce those methods, but instead, let's use `this.provider.request` directly

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @onhate! That's a great call. I've just proposed two style updates, and we should be good to go

File: index.js (hunk 2 above)
Comment (onhate) [reply index.js]: done

**Comment by onhate** [comment]
@medikoo I'll fix the tests

**Review by medikoo — APPROVED** [review]
Thank you @onhate ! That looks great

**Pull request merged**